examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,680 @@
+        return
+    }
+    displayedCastingPlayers.append(contentsOf: castingPlayers)
+}
+
+@objc
+func removeDiscoveredCastingPlayers(notification: Notification)
+{
+    guard let userInfo = notification.userInfo,
+    let castingPlayers = userInfo[castingPlayersUserInfo] as? [MTRCastingPlayer]
+    else {
+        print("No MTRCastingPlayer found in notification")
+        return
+    }
+    displayedCastingPlayers = displayedCastingPlayers.filter { !castingPlayers.contains($0)}
+}
+
+@objc
+func updateDiscoveredCastingPlayers(notification: Notification)
+{
+    guard let userInfo = notification.userInfo,
+    let castingPlayers = userInfo[castingPlayersUserInfo] as? [MTRCastingPlayer]
+    else {
+        print("No MTRCastingPlayer found in notification")
+        return
+    }
+    for castingPlayer in castingPlayers
+    {
+        if let index = displayedCastingPlayers.firstIndex(where: { castingPlayer.identifier == $0.identifier })
+        {
+            displayedCastingPlayers[index] = castingPlayer
+        }
+    }
+}
+```
+
+Finally, register these listeners and start discovery.
+
+On Android, call `addCastingPlayerChangeListener` on the singleton instance of
+`MatterCastingPlayerDiscoverer` to register the listener, and then call
+`startDiscovery`.
+
+```java
+public static void demoDiscovery()
+{
+    // Get the singleton instance of the MatterCastingPlayerDiscoverer
+    CastingPlayerDiscoverer discoverer = MatterCastingPlayerDiscoverer.getInstance();
+
+    // Add our castingPlayerChangeListener to listen to changes in the discovered CastingPlayers
+    discoverer.addCastingPlayerChangeListener(castingPlayerChangeListener);
+
+    // Start discovery
+    discoverer.startDiscovery();
+
+    // After some time, stop discovering and remove our castingPlayerChangeListener
+    // from the set of listeners the CastingPlayerDiscoverer informs
+    executorService.schedule(() -> {
+        discoverer.stopDiscovery();
+        discoverer.removeCastingPlayerChangeListener(castingPlayerChangeListener);
+    },     30, TimeUnit.SECONDS);
+}
+```
+
+On iOS, register the listeners by calling `addObserver` on the
+`NotificationCenter` with the appropriate selector, and then call `start` on the
+`sharedInstance` of `MTRCastingPlayerDiscovery`.
+
+```objectivec
+func startDiscovery() {
+    NotificationCenter.default.addObserver(
+    self,
+    selector: #selector(self.addDiscoveredCastingPlayers),
+    name: NSNotification.Name.didAddCastingPlayers,
+    object: nil)
+    NotificationCenter.default.addObserver(
+    self,
+    selector: #selector(self.removeDiscoveredCastingPlayers),
+    name: NSNotification.Name.didRemoveCastingPlayers,
+    object: nil)
+    NotificationCenter.default.addObserver(
+    self,
+    selector: #selector(self.updateDiscoveredCastingPlayers),
+    name: NSNotification.Name.didUpdateCastingPlayers,
+    object: nil)
+
+    MTRCastingPlayerDiscovery.sharedInstance().start()
+    self.discoveryRequestStatus = true
```

Comment:
What is discoveryRequestStatus needed for?

Reply:
This was just a state variable I waqs using to update a UI element on the sample app. I have replaced it with an "..." in the documentation to avoid confusing the reader.